src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -129,10 +130,22 @@ static ss::future<list_offset_partition_response> list_offsets_partition(
           offset,
           kafka_partition->leader_epoch());
     }
+    auto min_offset = kafka_partition->start_offset();
+    auto max_offset = model::prev_offset(offset);
+
+    // Empty partition.
+    if (max_offset < min_offset) {
+        co_return list_offsets_response::make_partition(
+          ktp.get_partition(),
+          model::timestamp(-1),
+          model::offset(-1),
+          kafka_partition->leader_epoch());
```

Comment:
Would it make sense to push this check into `cluster::partition` too / instead?

Reply:
We need to refactor this to use the bounded interval. Need to make it support Kafka offsets too first iirc.

Will merge it like this for now to make progress on the backports and will try to clean it up in a separate PR.

thanks for pointing it out!